src/Dynatrace.OpenTelemetry.Exporter.Metrics/LoggerExtensions.cs

```diff
@@ -0,0 +1,104 @@
+// <copyright company="Dynatrace LLC">
+// Copyright 2021 Dynatrace LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Net;
+using Microsoft.Extensions.Logging;
+using OpenTelemetry.Metrics;
+
+namespace Dynatrace.OpenTelemetry.Exporter.Metrics
+{
+	/// <summary>
+	/// Offers extensions around <see cref="ILogger"/> for high-performance logging.
+	/// <see href="https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/loggermessage?view=aspnetcore-6.0"/>
+	/// </summary>
+	internal static class LoggerExtensions
+	{
+		private static Action<ILogger, string, Exception> _dynatraceMetricUrl;
+		private static Action<ILogger, MetricType, Exception> _unknownMetricType;
+		private static Action<ILogger, HttpStatusCode, string, Exception> _sendMetricErrorResponse;
+		private static Action<ILogger, Exception> _sendMetricException;
+		private static Action<ILogger, string, Exception> _serializeMetricException;
+		private static Action<ILogger, string, Exception> _unsupportedMetricType;
+		private static Action<ILogger, string, Exception> _receivedCumulativeValue;
+
+		static LoggerExtensions()
```

Comment:
AFAIK the compiler adds static constructor checks to each static method if a static constructor is present.
Don't think the performance impact is too bad, but [this rule](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1810) describes it better than me probably.

Reply:
Good point. I removed the ctor and changed the fields to readonly.